process/block/preprocess/sovereignChainTransactions_test.go

```diff
@@ -608,4 +608,191 @@ func TestTxsPreprocessor_IsTransactionEligibleForExecutionShouldWork(t *testing.
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.EconomicsFee = &economicsmocks.EconomicsHandlerStub{
+			ComputeTxFeeCalled: func(tx data.TransactionWithFeeHandler) *big.Int {
+				return big.NewInt(1)
+			},
+		}
+
+		getRelayerWasCalled := false
+		args.TxProcessor = &testscommon.TxProcessorMock{
+			GetSenderAndReceiverAccountsCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, state2.UserAccountHandler, error) {
+				senderAccount := &state.UserAccountStub{
+					Nonce:   1,
+					Balance: big.NewInt(10),
+				}
+				return senderAccount, nil, nil
+			},
+			GetRelayerAccountCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, error) {
+				getRelayerWasCalled = true
+				return nil, nil
+			},
+		}
+
+		tp, _ := NewTransactionPreprocessor(args)
+		sctp, _ := NewSovereignChainTransactionPreprocessor(tp)
+
+		senderAddress := []byte("X")
+		tx := &transaction.Transaction{
+			SndAddr:          senderAddress,
+			Nonce:            1,
+			Value:            big.NewInt(2),
+			RelayerAddr:      senderAddress,
+			RelayerSignature: []byte("RS"),
+		}
+		err, value := sctp.isTransactionEligibleForExecution(tx, nil)
+		require.Nil(t, err)
+		require.True(t, value)
+
+		accntInfo, found := sctp.accntsTracker.getAccountInfo(senderAddress)
+		require.True(t, found)
+		require.Equal(t, uint64(2), accntInfo.nonce)
+		require.Equal(t, big.NewInt(7), accntInfo.balance)
+
+		require.False(t, getRelayerWasCalled)
+	})
+
+	t.Run("isTransactionEligibleForExecution should return true if relayed transaction", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.EconomicsFee = &economicsmocks.EconomicsHandlerStub{
+			ComputeTxFeeCalled: func(tx data.TransactionWithFeeHandler) *big.Int {
+				return big.NewInt(1)
+			},
+		}
+		args.TxProcessor = &testscommon.TxProcessorMock{
+			GetSenderAndReceiverAccountsCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, state2.UserAccountHandler, error) {
+				senderAccount := &state.UserAccountStub{
+					Nonce:   1,
+					Balance: big.NewInt(10),
+				}
+				return senderAccount, nil, nil
+			},
+			GetRelayerAccountCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, error) {
+				relayerAccount := &state.UserAccountStub{
+					Balance: big.NewInt(10),
+				}
+				return relayerAccount, nil
+			},
+		}
+
+		tp, _ := NewTransactionPreprocessor(args)
+		sctp, _ := NewSovereignChainTransactionPreprocessor(tp)
+
+		tx := &transaction.Transaction{
+			SndAddr:          []byte("X"),
+			Nonce:            1,
+			Value:            big.NewInt(2),
+			RelayerAddr:      []byte("R"),
+			RelayerSignature: []byte("RS"),
+		}
+		err, value := sctp.isTransactionEligibleForExecution(tx, nil)
+		require.Nil(t, err)
+		require.True(t, value)
+
+		accntInfo, found := sctp.accntsTracker.getAccountInfo(tx.GetSndAddr())
+		require.True(t, found)
+		require.Equal(t, uint64(2), accntInfo.nonce)
```

Comment:
Would it make sense here and below to check for sender nonce as well?

Reply:
Not sure what you mean. This is a check on sender nonce+balance and below there is a check for relayer balance.
If you meant relayer nonce, there is no need to check because it's not changed.